### ISSUE TITLE
Wrong groups of waypoints sorting in gpx

### ISSUE THREAD

**Issue by mariush444** [osmandapp/OsmAnd#20856]
### Description

It seems that sorting gruops of waypoint is wrong (not always the same).

Points in opend track are sorted OK. Alphabetically 
![o1](https://github.com/user-attachments/assets/7b054fef-8bb1-42ba-8274-7326a15de442)

Gruops to show on map are sorted OK.  Alphabetically 
![o2](https://github.com/user-attachments/assets/410545fc-e8fa-40d5-b54f-691362bbc901)

but ...
when you edit a waypoint in track then "select group" is sorted "some how - not alphabetically"
![w1](https://github.com/user-attachments/assets/5cfc283a-b527-4798-8846-e896e1c59c0c)

the same when you tap "list" button. It is sorted "some how - not alphabetically" (w2)
![w2](https://github.com/user-attachments/assets/da2cd832-97f0-43be-a442-bac81645c7b6)

### Steps to reproduce

as above

example of gpx is attatched
[gpx.zip](https://github.com/user-attachments/files/17023172/gpx.zip)

### Actual result

Sorting is not the same in every screen

### Expected result

Sorting should be always the same.
by me, prefered alphabetically

### Your Environment (required)

4.5.10 & 4.8.6

**Comment by yuriiurshuliak** [comment]
The issue has been reproduced. To reproduce it, import the provided track (SAU-iO-240904.gpx), then open the track menu and navigate to Points. You'll notice the waypoints are sorted alphabetically.

Next, select any waypoint on the map, tap it, choose Actions, and then select Edit GPX waypoint. In the group selection tab, you will see that the groups are displayed in a random order.

OsmAnd~ 4.9.0#3203m, released: 2024-09-17

<table style="width:100%">
  </tr>
  <tr>
    <th>Alphabetical sorting</th>
        <th>Alphabetical sorting</th>
    <th>Chaotic sorting</th>
<th>Chaotic sorting</th>
  </tr>
  <tr>
<th><img src="https://github.com/user-attachments/assets/6be5e353-28d8-4fbb-88c2-6ec23a19ae16"  width="300" height="auto"/> </th>
 <th><img src="https://github.com/user-attachments/assets/15d30fe7-c1fe-4917-9e75-9357ef171795"  width="300" height="auto"/> </th>
 <th><img src="https://github.com/user-attachments/assets/86082de3-bdd4-4d61-8932-5e0cf3c319dc"  width="300" height="auto"/> </th>
  <th><img src="https://github.com/user-attachments/assets/0fd699ed-67b2-4b21-abf6-85b4a7074da0"  width="300" height="auto"/> </th>
  </tr>
</table>

**Comment by vshcherb** [comment]
On these screens "last used" is used, so it works as intented.

**Comment by mariush444** [comment]
Can user use alphabetical as on other screens? 
it is more readable and uniform

**Comment by vshcherb** [comment]
Nope, categories are done on these screens so it's easy to access to last used. However we will double check the relevance

**Comment by mariush444** [comment]
Yes, please reconsider it 
Even at the 1st modofication the list is not sorted (is unsorted at all)

**Comment by vshcherb** [comment]
3rd screen will be replaced with new one and proper sorting, horizontal screen is sorted by usage for easy adding to last used group
